utils/cache/filecache/filecache.go

```diff
@@ -0,0 +1,210 @@
+package filecache
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+)
+
+type Cache struct {
+	entries       iEntryMap
+	entriesLM     iLockMap
+	entryProvider IEntryRetriever
+	fs            filesystem.FS
+	cfg           *FileCacheConfig
+	cancelStore   *parallelisation.CancelFunctionStore
+	closed        atomic.Bool
+}
+
+func (c *Cache) gc(ctx context.Context, _ time.Time) {
+	if err := parallelisation.DetermineContextError(ctx); err != nil {
+		return
+	}
+
+	c.entries.Range(func(key string, entry ICacheEntry) {
+		if entry.IsExpired() && c.entriesLM.TryLock(key) {
+			if err := entry.Delete(ctx); err == nil {
+				c.entries.Delete(key)
+				defer func() {
+					c.entriesLM.Unlock(key)
+					c.entriesLM.Delete(key)
+				}()
+			}
+		}
+	})
+}
+
+func (c *Cache) checkIfClosed() error {
+	if c.closed.Load() {
+		return commonerrors.New(commonerrors.ErrConflict, "cache is closed")
+	}
+
+	return nil
+}
```

Comment:
```suggestion
func (c *Cache) closed() bool {
	return c.closed.Load()
}
```

I don't think this should return an error type. To me, and error implied an issue occurred when the function ran, but in here the entire point of the function is to check if the cache is closed.

so if the cache is closed then the function has done it's job correctly and it shouldn't return an error.

I would then return an error when you do things like:

```go
    if c.closed() {
		return false, commonerrors.New(commonerrors.ErrConflict, "cache is closed")
	}
	```

Reply:
the purpose of the function is to wrap the error, if this need to change, then we have one place to maintain this, this error is supposed to be in every cache operation, that why I wanted to refactor that in a separate function